File: Miscellaneous/Golang/242. Valid Anagram.go
package main

import "fmt"

func main() {

	s, t := "anagram", "nagaram"
	result := isValidAnagram(s, t)
	fmt.Println(result)
}

func isValidAnagram(str1, str2 string) bool {

	if len(str1) != len(str2) {
		return false
	}

	count := 0

	for i := 0; i < len(str1); i++ {
		count += int(str1[i])
	}

	for i := 0; i < len(str2); i++ {
		count -= int(str2[i])
	}

	if count == 0 {
		return true
	}
	return false
}

File: Miscellaneous/Python/242. Valid Anagram.py
class Solution:
    def isValidAnagram(self, str1, str2):

        if len(str1) is not len(str2):
            return False

        count = 0

        for char in str1:
            count += ord(char)

        for char in str2:
            count -= ord(char)

        if count == 0:
            return True
        return False

obj = Solution()
s, t = "anagram", "nagaram"
ans = obj.isValidAnagram(s, t)
print(ans)